akka-javasdk/src/main/java/akka/javasdk/client/ComponentMethodRef1.java

```diff
@@ -22,5 +23,21 @@ public interface ComponentMethodRef1<A1, R> extends ComponentDeferredMethodRef1<
 
   ComponentMethodRef1<A1, R> withMetadata(Metadata metadata);
 
+  /**
+   * Set the retry settings for this call.
+   *
+   * @param retrySettings The retry settings
+   * @return A new call with the retry settings set
+   */
+  ComponentMethodRef1<A1, R> withRetry(RetrySettings retrySettings);
```

Comment:
this should return ComponentInvokeOnlyMethodRef1

Reply:
yeah, good idea